ja/lessons/advanced/otp-supervisors.md

```diff
@@ -1,95 +1,163 @@
 ---
-version: 1.0.1
-title: OTP スーパーバイザ
+version: 1.1.1
+title: OTP スーパバイザ
 ---
 
-Supervisor(スーパーバイザ)は、他のプロセスを監視するという、1つの目的に特化したプロセスです。子プロセスが失敗した際に自動的に再起動させることによって、耐障害性の高い(フォールトトレラントな)アプリケーションを作ることを可能にします。
+スーパバイザは、他のプロセスを監視するという一つの目的に特化したプロセスです。
+子プロセスが失敗した際に自動的に再起動させることによって、耐障害性の高いアプリケーションを作ることを可能にします。
 
 {% include toc.html %}
 
 ## 設定
 
-Supervisor の魔術は `Supervisor.start_link` 関数の中にあります。スーパーバイザプロセスと子プロセスを開始するのに加えて、子を管理するためにスーパーバイザが使用する戦略を定義することができます。
+スーパバイザの魔術は `Supervisor.start_link/2` 関数の中にあります。
+[OTP の並行性](../../advanced/otp-concurrency)レッスンで実装した SimpleQueue を用いて、始めていきましょう。
 
-子プロセスはリストと、 `Supervisor.Spec` から取り込まれた `worker/3` 関数を使って定義されます。 `worker/3` 関数はモジュール、引数、そして一連のオプションを受け取ります。内部では、 `worker/3` は初期化時に、与えられた引数を用いて `start_link/3` を呼び出します。
+新しいプロジェクトを `mix new simple_queue --sup` を使って作成することで、スーパバイザツリーも一緒に作成することができます。
+`SimpleQueue` モジュール用のコードは `lib/simple_queue.ex` へ、スーパバイザを起動するコードは `lib/simple_queue/application.ex` に記述することになります。
 
-[OTP の並行性](../../advanced/otp-concurrency)レッスンで実装した SimpleQueue を用いて、始めていきましょう:
+スーパバイザの子プロセスはリストで定義します。それは以下のように単純にモジュール名のリストにするか…
 
 ```elixir
-import Supervisor.Spec
+defmodule SimpleQueue.Application do
+  use Application
+
+  def start(_type, _args) do
+    children = [
+      SimpleQueue
+    ]
+
+    opts = [strategy: :one_for_one, name: SimpleQueue.Supervisor]
+    Supervisor.start_link(children, opts)
+  end
+end
+```
 
-children = [
-  worker(SimpleQueue, [], name: SimpleQueue)
-]
+あるいは子プロセスの起動時に初期値を渡すべくタプルのリストにするか、のどちらかが使えます。
 
-{:ok, pid} = Supervisor.start_link(children, strategy: :one_for_one)
+```elixir
+defmodule SimpleQueue.Application do
+  use Application
+
+  def start(_type, _args) do
+    children = [
+      {SimpleQueue, [1, 2, 3]}
+    ]
+
+    opts = [strategy: :one_for_one, name: SimpleQueue.Supervisor]
+    Supervisor.start_link(children, opts)
+  end
+end
+```
+
+`iex -S mix` で実行すると `SimpleQueue` が自動的に実行されるのがわかります。
+
+```elixir
+iex> SimpleQueue.queue
+[1, 2, 3]
 ```
 
-プロセスが停止したり終了させられたりすれば、スーパーバイザは何事もなかったかのように自動的にプロセスを再起動するでしょう。
+もし `SimpleQueue` プロセスが異常終了や通常終了すれば、スーパバイザは何事もなかったかのように自動的にプロセスを再起動するでしょう。
 
 ### 戦略
 
-今のところ、スーパーバイザが利用可能な4つの異なる再起動の戦略があります:
+
```

Comment:
empty line :eyes:

Reply:
I do not care much about that. If it causes a different output it should be removed.